src/parser/flags.ts

```diff
@@ -35,8 +35,10 @@ export function boolean<T = boolean>(
   } as BooleanFlag<T>
 }
 
-export const integer = (opts: { min?: number; max?: number} & Partial<OptionFlag<number>> = {}): OptionFlag<number | undefined> => {
-  return build<number>({
+export function integer(opts: Partial<OptionFlag<number>> & {min?: number; max?: number } & ({required: true} | { default: number })): OptionFlag<number>
+export function integer(opts: Partial<OptionFlag<number>> & {min?: number; max?: number }): OptionFlag<number | undefined>
+export function integer(opts: Partial<OptionFlag<number>> & {min?: number; max?: number }): OptionFlag<number> | OptionFlag<number | undefined> {
```

Comment:
Why is the duplicated `OptionFlag<T>` return type needed here?
```suggestion
export function integer(opts: Partial<OptionFlag<number>> & {min?: number; max?: number }): OptionFlag<number | undefined> {
```